.github/workflows/portable_linux_package_matrix.yml

```diff
@@ -50,14 +53,14 @@ jobs:
       DIST_ARCHIVE: "${{ github.workspace }}/output/therock-dist-${{ matrix.target_bundle.amdgpu_family }}${{ github.events.input.package_suffix }}.tar.gz"
     steps:
       - name: "Checking out repository"
-        uses: actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3 # v3.5.0
-      - uses: actions/setup-python@v5
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - uses: actions/setup-python@42375524e23c412d93fb67b49958b491fce71c38 # v5.4.0
         with:
           python-version: '3.10'
       # TODO: We shouldn't be using a cache on actual release branches, but it
       # really helps for iteration time.
       - name: Enable cache
-        uses: actions/cache/restore@v4
+        uses: actions/cache/restore@0c907a75c2c80ebcb7f088228285e798b750cf8f # v4.2.1
```

Comment:
Want to update the other instances of this too?

https://github.com/nod-ai/TheRock/blob/3d579ae7ce21ebe203484b96175aab57ef8032c8/.github/workflows/build_linux_packages.yml#L47-L48

https://github.com/nod-ai/TheRock/blob/3d579ae7ce21ebe203484b96175aab57ef8032c8/.github/workflows/build_windows_packages.yml#L69-L70

https://github.com/nod-ai/TheRock/blob/3d579ae7ce21ebe203484b96175aab57ef8032c8/.github/workflows/build_python_packages.yml#L52-L53

(and for `actions/cache/save`)

Reply:
I was hesitant as I think those workflows will be more or less entirely replaced. But it's probably worth it to get us Dependabot earlier. Will do this in a follow up.